core/cat/plugins/quickstart_cat_plugin/models.py

```diff
@@ -72,20 +72,27 @@ def get_language_model(cat):
 
 @hook
 def get_language_embedder(cat):
-    # TODO: give more example ocnfigurations
+    # TODO: give more example configurations
 
     # Embedding LLM
-    using_openai_llm = type(cat.llm) in [OpenAI, OpenAIChat]
+    using_openai_llm = type(cat.llm) in [OpenAI, OpenAIChat, AzureOpenAI]
     if ("OPENAI_KEY" in os.environ) or using_openai_llm:
         openai_key = os.getenv("OPENAI_KEY")
         if openai_key is None:
             openai_key = cat.llm.openai_api_key
-        embedder = embedders.EmbedderOpenAIConfig.get_embedder_from_config(
-            {
-                "openai_api_key": openai_key,
-                # model_name: '....'  # TODO: allow optional kwargs
-            }
-        )
+        if using_openai_llm in [OpenAI, OpenAIChat]:
```

Comment:
@pieroit should I reverse the logic here ? Check of `AzureOpenAI` and use the embedder `EmbedderAzureOpenAIConfig` and then have a catch all `else` ?

Reply:
@zioproto @soferreira I made some refactoring to avoid confusion between default cat behaviour and plugin extensions.

If you look into `cat/mad_hatter/core_plugin` you will find some hooks that define how the cat loads LLM and embedder.

You can override the `get_language_model` and `get_language_embedder` hooks to always have the cat loading a specific LLM/embedder. In `cat/plugins/myplugin/myfile.py`:

```python

@hook
def get_language_model(cat):

    # look here in DB, or totally skip it

    config = {
        openai_api_key: os.environ["SOME_ENV_VAR"]
        api_base: "something"
        api_type: "azure"
        api_version: "2022-12-01"
    }

    return LLMAzureOpenAIConfig.get_llm_from_config(config)

```

From now on your hook will be executed and not the default one.
Hope this is intuitive enough, feedback is welcome